bootstrap.go

```diff
@@ -31,11 +37,12 @@ func GracefulDatadogShutdown() {
 	defer profiler.Stop()
 }
 
-func initTracer(cfg config.DatadogParameters, isConnectionSocket bool) {
+func initTracer(cfg config.DatadogParameters, connectionType ConnectionType) {
 	var tracerOptions []tracer.StartOption
-	if isConnectionSocket {
+	switch connectionType {
```

Comment:
Not sure why, but `If` will be faster than `Switch` since Go uses binary search, so for big scale its better but for 2 cases.....it's like a shooting cow with rocket launcher

Reply:
In this particular case I tend to agree that the switch is cleaner. Since we are running this once at startup, runtime complexity is not that important.

Reply:
@stephendrake @nhhagen For readability, I agree but less likely Datadog will add another protocol as a connection... Your changing forces consumers of pkg change code as well so the value of doing this "refactoring" is almost zero